app/views/devise/registrations/edit.html.erb

```diff
@@ -3,9 +3,29 @@
 <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
   <%= render "devise/shared/error_messages", resource: resource %>
 
+  <div class="field">
+    <%= f.label :firstname %><br />
+    <%= f.email_field :firstname, autofocus: true, autocomplete: "email" %>
```

Comment:
Do we need `autocomplete: "email"`?

Reply:
we don't need it! I just added it for ease.